### DISCUSSION TITLE
Use global .env file

### DISCUSSION THREAD

**Discussion by ThibautCuchet** [vercel/turborepo#1676] [Help] [answered]
Hi everyone,

I'm new to Turborepo and I have troubles using .env file in my project.
The structure of my monorepo is :

- apps
    - next_app_1
    - next_app2
- packages
    - hasura
    - ui

There are commons env variable between next_app_1, next_app_2 and hasura used in ts files and configuration (like yml). My project works when I put an env file in each sub-folder but I prefer to have one .env file at the root folder to edit them easily.

Is someone could help me ?

**Comment by EarlGeorge** [comment, marked as answer]
https://turborepo.org/blog/turbo-1-1-0#environment-variable-dependencies

**Reply by ThibautCuchet** [reply]
Thanks for the link but I already tried in my repository and it doesn't change anything

**Reply by EarlGeorge** [reply]
I am not sure about your setup, so you need to put an effort to figure it out, unless if someone else can try to help you here, also better would be not to use global env for monorepo

**Comment by mehulkar** [comment]
As far as Turborepo is concerned, @EarlGeorge is correct, you can declare the "global" environment variable _dependencies_ in `turbo.json` in the `globalDependencies` config. This config tells Turborepo that when these environment variables change, Turborepo should invalidate its cache.

*Using* a `.env` file from the root of your monorepo in all your apps depends on how your project is setup. This may be a better question for Stackoverflow or  you could try asking for general help in the [Turborepo Discord server](https://turborepo.org/discord).
